internal/test/scorecard/tests.go

```diff
@@ -288,3 +288,89 @@ func logEvents(r *scapiv1alpha3.TestResult, client *CryostatClientset, namespace
 	}
 	return nil
 }
+
+func newCryostatCR(namespace string, withIngress bool) *operatorv1beta1.Cryostat {
+	cr := &operatorv1beta1.Cryostat{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cryostat-cr-test",
+			Namespace: namespace,
+		},
+		Spec: operatorv1beta1.CryostatSpec{
+			Minimal:           false,
+			EnableCertManager: &[]bool{true}[0],
+		},
+	}
+
+	if withIngress {
+		pathType := v1.PathTypePrefix
+		cr.Spec.NetworkOptions = &operatorv1beta1.NetworkConfigurationList{
+			CoreConfig: &operatorv1beta1.NetworkConfiguration{
+				Annotations: map[string]string{
+					"nginx.ingress.kubernetes.io/backend-protocol": "HTTPS",
+				},
+				IngressSpec: &v1.IngressSpec{
+					TLS: []v1.IngressTLS{},
+					Rules: []v1.IngressRule{
+						{
+							Host: "testing.cryostat",
```

Comment:
I think that value is passed in by the user when they configure Ingress spec in Cryostat CR. 

I just used the value from [config.md](https://github.com/cryostatio/cryostat-operator/blob/main/docs/config.md#network-options)

Reply:
tho the service name is incorrect as it needs to be `CRName-...`. I will update that.

Reply:
I see, that's sort of what I thought. I think this should somehow be configurable then in case we need to use some different ingress host in different testing environments.

Reply:
Make sense to me! Just that i am not sure there is a nice way to do that with scorecard. I think we could specify flags and arguments in scorecard config.

Reply:
That part I'm not sure about, I don't know the scorecard tests well enough, but it sounds feasible.

Reply:
I'm not sure if these Ingress will work with Kind and Scorecard: https://github.com/kubernetes-sigs/kind/issues/1339.

Reply:
This might work: https://stackoverflow.com/a/66982597. Expose the Ingress Controller as a Service and then use the service URL while specifying the `Host` header with whatever we choose for our Ingress hosts.

Reply:
This should be a prereq for the cluster right? Then we need to find the ingress service and grab its IP?

Reply:
> This issue affects testing out Cryostat API, right?

Right. I don't believe it will affect what the tests do currently.

> This should be a prereq for the cluster?
> 
> Then we need to find the ingress service and grab its IP within scorecard test?

Right. The Service hostname might work instead, if the `Host` header takes priority. It would be slightly easier than retrieving the IP. This URL could be passed to the scorecard test as an optional argument, like the existing cert-manager one. I don't see any RBAC concerns with this, but maybe I'm missing something.

> Or I am wondering if we can trust ingress rule and controller is behaving "correctly" and just make request directly to service for Cryostat server?

That is one option, but it would depend on https://github.com/cryostatio/cryostat-operator/issues/403. That's non-trivial to implement though.

Reply:
Any other comments for this..??

Reply:
Still trying to test this in conjunction with your other PR. Did you happen to have deleted your test bundle image?
`ghcr.io/tthvo/cryostat-operator-bundle:ci-abc78d7db8a6080f9bbe602c01e93e60970f8733`

I also want to verify that the `Host` header + service workaround works for the scorecard tests.

Reply:
Just added `read` access :))

Reply:
The Host workaround seems to work. Additionally, there was already a service created for ingress-nginx. This is from the operator pod:
```
$ curl -k -H "Host: testing.cryostat" https://ingress-nginx-controller.ingress-nginx.svc/health
{"reportsAvailable":true,"reportsConfigured":false,"datasourceConfigured":true,"dashboardAvailable":true,"datasourceAvailable":true,"cryostatVersion":"v2.4.0-dev-32-gf9a7f00f","dashboardConfigured":true}
$ curl -k -H "Host: testing.cryostat-grafana" https://ingress-nginx-controller.ingress-nginx.svc/api/health
{
  "commit": "4add91f03d",
  "database": "ok",
  "version": "9.4.7"
}
```

This successfully queries the Cryostat and Grafana health endpoints.

Reply:
Oh very niceee!!